wgpu-hal/src/metal/device.rs

```diff
@@ -1272,3 +1312,22 @@ impl crate::Device for super::Device {
         unimplemented!()
     }
 }
+
+// TODO: `label` is not exposed on `MTLBuffer`, is this always correct?
+fn set_buffer_label(buffer: &ProtocolObject<dyn MTLBuffer>, label: Option<&NSString>) {
+    unsafe { msg_send![buffer, setLabel: label] }
+}
+
+// TODO: `label` is not exposed on `MTLTexture`, is this always correct?
+fn set_texture_label(buffer: &ProtocolObject<dyn MTLTexture>, label: Option<&NSString>) {
+    unsafe { msg_send![buffer, setLabel: label] }
+}
```

Comment:
They both inherit from [`MTLResource`](https://developer.apple.com/documentation/metal/mtlresource?language=objc) which has a `label`.

Reply:
Right, I didn't think to import that, fixed now